manifests/fact.pp

```diff
@@ -6,25 +6,33 @@
   $value,
   String $fact                    = $name,
   String $file                    = 'facts.txt',
-  Stdlib::Absolutepath $facts_dir = '/etc/facter/facts.d',
+  Optional[Stdlib::Absolutepath] $facts_dir = undef,
```

Comment:
why allow undef?

Reply:
The alternative is a switch statement in the parameters to change default based on default.  I opted for undef so that `pick()` could be used to take default from `facter` class.